### PR TITLE
Add a `delete` option to ImapFolderAgent

### PR DESCRIPTION
**Pull request by knu**
In addition to `mark_as_read`, ImapFolderAgent should be able to delete found mails.

### CODE REVIEW

File: app/models/agents/imap_folder_agent.rb [hunk 1]
```diff
@@ -315,6 +316,11 @@ def check
           log 'Marking as read'
           mail.mark_as_read unless dry_run?
         end
+
+        if boolify(interpolated['delete'])
+          log 'Deleting'
```
Comment (dsander) [inline app/models/agents/imap_folder_agent.rb]: Would it make sense to log the `message_id` or something more human readable here?

Comment (knu) [reply app/models/agents/imap_folder_agent.rb]: Yeah, but unless multiple jobs of an agent are run simultaneously, you can tell which mail is read or deleted by the earlier log line that should be emitted. (There's this line: `log 'Emitting an event for mail: %s' % message_id`)

Maybe AgentLog should have a worker ID (instance ID + PID/TID?) when we officially introduce parallelism in Huginn workers. 😄

**Comment by knu** [comment]
@dsander Thanks for taking a look!

File: app/models/agents/imap_folder_agent.rb (hunk 1 above)
Comment (dsander) [reply app/models/agents/imap_folder_agent.rb]: Gotcha, did not notice that the lines above were always used.

In theory we already have parallelism when multiple DJ workers are used but for the imap agent that should not be an issue.

**Pull request merged**